src/_pytest/main.py

```diff
@@ -394,14 +403,16 @@ def __init__(self, config):
         self.trace = config.trace.root.get("collection")
         self._norecursepatterns = config.getini("norecursedirs")
         self.startdir = config.invocation_dir
-        self._initialpaths = frozenset()
+        self._initialpaths = frozenset()  # type: FrozenSet[py.path.local]
+
         # Keep track of any collected nodes in here, so we don't duplicate fixtures
-        self._node_cache = {}
+        self._node_cache = {}  # type: Dict[str, List[Node]]
```

Comment:
Apparently sometimes the key can be tuple so the type is not exactly correct:

https://github.com/pytest-dev/pytest/blob/ad02f6f8790df4252ba9af69aa76c58eb105a1db/src/_pytest/main.py#L574-L579

I am working on some annotations and will include a fix for this, hopefully tomorrow.